dbt_meshify/storage/file_content_editors.py

```diff
@@ -0,0 +1,453 @@
+    """Processes the yml contents to be written back to a file"""
+    model_ordered_dict = OrderedDict.fromkeys(
+        [
+            "name",
+            "description",
+            "latest_version",
+            "access",
+            "group",
+            "config",
+            "meta",
+            "tests",
+            "columns",
+            "versions",
+        ]
+    )
+    model_ordered_dict.update(model_yml)
+    # remove any keys with None values
+    return filter_empty_dict_items(model_ordered_dict)
+
+
+def resources_yml_to_dict(resources_yml: Dict, resource_type: NodeType = NodeType.Model):
+    """Converts a yml dict to a named dictionary for easier operation"""
+    return (
+        {resource["name"]: resource for resource in resources_yml[resource_type.pluralize()]}
+        if resources_yml
+        else {}
+    )
+
+
+class DbtMeshFileEditor:
+    """
+    Class to operate on the contents of a dbt project's files
+    to add the dbt mesh functionality
+    includes editing yml entries and sql file contents
+    """
+
+    @staticmethod
+    def add_group_to_yml(group: Group, groups_yml: Dict[str, Any]):
+        """Add a group to a yml file"""
+        if groups_yml is None:
+            groups_yml = {}
+
+        groups = resources_yml_to_dict(groups_yml, NodeType.Group)
+        group_yml = groups.get(group.name) or {}
+
+        group_yml.update({"name": group.name})
+        owner = group_yml.get("owner", {})
+        owner.update(filter_empty_dict_items(group.owner.to_dict()))
+        group_yml["owner"] = owner
+
+        groups[group.name] = filter_empty_dict_items(group_yml)
+
+        groups_yml["groups"] = list(groups.values())
+        return groups_yml
+
+    @staticmethod
+    def add_access_to_model_yml(
+        model_name: str, access_type: AccessType, models_yml: Dict[str, Any]
+    ):
+        """Add group and access configuration to a model's YAMl properties."""
+        # parse the yml file into a dictionary with model names as keys
+        models = resources_yml_to_dict(models_yml)
+        model_yml = models.get(model_name) or {"name": model_name, "columns": [], "config": {}}
+
+        model_yml.update({"access": access_type.value})
+        models[model_name] = process_model_yml(model_yml)
+
+        models_yml["models"] = list(models.values())
+        return models_yml
+
+    @staticmethod
+    def add_group_to_model_yml(model_name: str, group: Group, models_yml: Dict[str, Any]):
+        """Add group and access configuration to a model's YAMl properties."""
+        # parse the yml file into a dictionary with model names as keys
+        models = resources_yml_to_dict(models_yml)
+        model_yml = models.get(model_name) or {"name": model_name, "columns": [], "config": {}}
+
+        model_yml.update({"group": group.name})
+        models[model_name] = process_model_yml(model_yml)
+
+        models_yml["models"] = list(models.values())
+        return models_yml
```

Comment:
This method, `add_access_to_model_yml`, and `add_group_to_yml` are _very_ similar in function -- specifically they're adding properties to a specific model property entry. This is fine for now, since this is a first draft. Having that been said, this might be something to add to the IO refactoring work I've been pondering.

Reply:
these were copy pasted in order to access each method separately, rather than only as part of `add_access_and_group_to_yml` -- I think the steps in all of these methods are overlapping and too verbose -- i want to shrink this file dramatically!